packages/api/amplify_api/lib/src/graphql/web_socket/state/web_socket_state.dart

```diff
@@ -114,11 +191,76 @@ class ConnectedState extends WebSocketState {
     super.intendedState,
     super.service,
     super.subscriptionBlocs,
+    super.options,
     this.timeoutTimer,
+    this.pollTimer,
   );
 
   /// timer used for missed ka messages
   final RestartableTimer timeoutTimer;
+
+  /// timer used for periodic polling
+  final Timer pollTimer;
+
+  /// Move state to [ConnectingState]
+  @override
+  ConnectingState connecting({
+    NetworkState? networkState,
+    IntendedState? intendedState,
+  }) {
+    _cancelTimers();
+
+    return super.connecting(
+      networkState: networkState ?? this.networkState,
+      intendedState: intendedState ?? this.intendedState,
+    );
+  }
+
+  /// Move state to [ReconnectingState]
+  @override
+  ReconnectingState reconnecting({
+    NetworkState? networkState,
+    IntendedState? intendedState,
+  }) {
+    _cancelTimers();
+
+    return super.reconnecting(
+      networkState: networkState ?? this.networkState,
+      intendedState: intendedState ?? this.intendedState,
+    );
+  }
+
+  /// Move state to [DisconnectedState]
+  @override
+  DisconnectedState disconnect() {
+    _cancelTimers();
+
+    return super.disconnect();
+  }
+
+  /// Move state to [FailureState]
+  @override
+  FailureState failed(
+    Object e, {
+    StackTrace? st,
```

Comment:
Can make required. If the stack trace is unavailable (because you've instantiated an exception instead of catching a thrown one), you can use `StackTrace.current`